Submission of article on Azure Key Vault

joydipkanjilal opened the pull request:
##### Description of change

Submission of article on Azure Key Vault

##### Author Details for Blog Post Entry

id: Joydip Kanjilal
bio: A Microsoft Most Valuable Professional in ASP.NET, Speaker, and Author of several books and articles. More than 25 years of experience in IT with more than 18 years in Microsoft .NET and its related technologies. Authored 8 books and more than 500 articles in some of the most reputed sites worldwide including MSDN, Info World, CodeMag, Tech Beacon, Tech Target, etc.
github: https://github.com/joydipkanjilal
linkedin: https://in.linkedin.com/in/joydipkanjilal
medium: https://joydipkanjilal.medium.com/
twitter: https://twitter.com/joydipkanjilal

puneetsingh24 commented on content/working-with-azure-key-vault-in-csharp/Working with Azure Key Vault in CSharp.md:
````diff
@@ -0,0 +1,451 @@
+```
+---
+title: **Working with Azure Key Vault in C\#**
````
Remove asterisk from the title, use can use double quotes instead

joydipkanjilal replied:
Hi @puneetsingh24 I've made all the changes and checked in the code.

puneetsingh24 commented on content/working-with-azure-key-vault-in-csharp/index.md:
````diff
@@ -0,0 +1,455 @@
+---
+```
+
+## **Introduction**
````
Remove the asterisk from all the headings, they are by default bold

joydipkanjilal replied:
Done, thanks!

puneetsingh24 commented on content/working-with-azure-key-vault-in-csharp/index.md:
```diff
@@ -0,0 +1,455 @@
+
+2.  Select Web App from the list
+
+![](media/881710611150de6c542f6656de3b727e.png)
```
Add alt text in all the images in the blog [alt text](image path)

joydipkanjilal replied:
Done, thanks!

puneetsingh24 reviewed (changes requested):
Hello @joydipkanjilal , please do the following changes

Please Add author info under content->author.yaml file
Remove the extra file - Working with Azure Key Vault in CSharp.md
Remove the asterisk from all the headings
Add the alt text in all the images in the post

raghunath-r-a commented on content/working-with-azure-key-vault-in-csharp/index.md:
```diff
@@ -56,14 +56,14 @@ To execute the code examples provided in this article you should have each of th
-## **Understanding Managed Identity**
+## Understanding Managed Identity
 
 Managed identity is a concept that eliminates the need of having to store credentials once an application has been deployed to the cloud. By using managed identity, we can securely access a variety of Azure services without having to store any credentials like as connection strings or passwords. Managed identity may be used to connect to Key Vault from an Azure Function App or an Azure Web App, a ...
```
Changes: Managed identity is a concept that eliminates the need to store credentials once an application has been deployed to the cloud. (deleted `of having`)

The second highlighted part is repeating the same fact stated in the previous sentence: ``` it helps ``` (*this part is edited*) securely access a variety of Azure services `without having to store any credentials like as connection strings or passwords`.

Changes: `You can also use m`anaged identity to connect to Key Vault from an Azure Function App or an Azure Web App, as well as to connect to Azure Blob Storage from an Azure Web App.

Changes: Managed identities are two types: (deleted  `of the following`)

raghunath-r-a commented on content/working-with-azure-key-vault-in-csharp/index.md:
```diff
@@ -73,7 +73,7 @@ Managed identities are of the following two types:
-## **Create a Web App in the Azure Portal**
+## Create a Web App in the Azure Portal
 
 We will now create an Azure Web App instance with the permissions to access Azure Key Vault. Adhere to the steps given below that would guide you to create a new Azure Web App instance while you’re within the Azure portal:
```
Please use contractions consistently throughout the blog. For example, `We will` vs. `We'll`; `You are` vs. `You're`

raghunath-r-a replied:
Here, `alt-text` is missing for the images. Please add relevant `alt-text`, which should describe the image in a in 3 to 5 words.

raghunath-r-a replied:
Use `--` to create an em dash here: In the Publish section, there are two choices, Code and Docker Container `-` choose Code

raghunath-r-a replied:
Set the `operating` system to Windows

raghunath-r-a commented on content/working-with-azure-key-vault-in-csharp/index.md:
```diff
@@ -117,11 +117,11 @@ To enable the system-assigned managed identity for the Azure Web App we just cre
-## **Create and Configure Azure Key Vault**
+## Create and Configure Azure Key Vault
 
 In this section we’ll examine how we can create and configure an Azure key Vault instance.
```
Changes: In this section`,` we’ll examine how `to` create and configure an Azure key Vault instance.

raghunath-r-a commented on content/working-with-azure-key-vault-in-csharp/index.md:
```diff
@@ -117,11 +117,11 @@ To enable the system-assigned managed identity for the Azure Web App we just cre
-### **Create a new Azure Key Vault Instance**
+### Create a new Azure Key Vault Instance
 
 We'll now create an Azure Key Vault in the Azure Portal and then add a secret to it. To create a new Azure Key Vault instance, navigate to the Azure Portal and follow the steps mentioned below:
```
Changes: follow the steps below: (deleted: `mentioned`)

raghunath-r-a replied:
Changes: In the Create key vault screen`,` provide the name, subscription, resource group name, location`,` and leave the other options to their default values.

raghunath-r-a commented on content/working-with-azure-key-vault-in-csharp/index.md:
```diff
@@ -147,7 +147,7 @@ Figure 5
-### **Create an App Secret in Azure Key Vault**
+### Create an App Secret in Azure Key Vault
 
 Now that a Key Value instance has been created, we'll add a secret to it and then make the secret accessible to the Azure Web App we created earlier. Follow the steps mentioned below to create a secret:
```
Changes: Follow the steps below. (Deleted `mentioned`)

raghunath-r-a commented on content/working-with-azure-key-vault-in-csharp/index.md:
```diff
@@ -169,7 +169,7 @@ Now that a Key Value instance has been created, we'll add a secret to it and the
-### **Authorize the Web App to access your Key Vault**
+### Authorize the Web App to access your Key Vault
 
 To provide access to the secret we created, follow the steps mentioned below:
```
Use either `you` or `we` consistently throughout the article. Please don't use both, which would cause inconsistency and may confuse readers.

raghunath-r-a replied:
Also, just saying `steps below` would suffice. No need to say `steps mentioned below`.

joydipkanjilal replied:
I've made the changes, thanks!

joydipkanjilal replied:
> 
> 
> Use either `you` or `we` consistently throughout the article. Please don't use both, which would cause inconsistency and may confuse readers.

I've used "you" throughout this article.

raghunath-r-a commented on content/working-with-azure-key-vault-in-csharp/index.md:
```diff
@@ -192,7 +192,7 @@ To provide access to the secret we created, follow the steps mentioned below:
-## **Create a new ASP.NET 5 Core application**
+## Create a new ASP.NET 5 Core application
 
 First of all, you will create an ASP.NET Core 5 web application. The project type comes bundled with all the template files to create a web application, even before you add something. Follow the steps given below to create a new ASP.NET Core Web application within the Visual Studio 2019 IDE.
```
Either end points with a pull stop or doesn't use a pull stop. But please make sure usage of pull stop is consistent through the articles wherever points are present. Our recommendation is not to use a pull stop in points.

joydipkanjilal replied:
Fixed, thanks!

raghunath-r-a commented on content/working-with-azure-key-vault-in-csharp/index.md:
```diff
@@ -235,11 +235,11 @@ You may install these packages in one of two ways: either via the NuGet Package
-#### **Specify the Vault Uri in AppSettings**
+#### Specify the Vault Uri in AppSettings
 
 Create a section named KeyVault in the appsettings.json file and specify a key
```
Wherever there is a file, please highlight it with ``

Here, `appsettings.json`

joydipkanjilal replied:
Done, thanks!

raghunath-r-a commented on content/working-with-azure-key-vault-in-csharp/index.md:
```diff
@@ -410,7 +410,7 @@ services.AddControllersWithViews();
-## **Deploy the Application to Azure**
+## Deploy the Application to Azure
 
 To deploy the application, follow the steps mentioned below:
```
Are ellipses necessary here: Select Publish`...`

raghunath-r-a replied:
Changes: Specify, or associate`,` the App Service instance with your application

joydipkanjilal replied:
> 
> 
> Are ellipses necessary here: Select Publish`...`

No, I've removed it.

raghunath-r-a commented on content/working-with-azure-key-vault-in-csharp/index.md:
```diff
@@ -438,15 +438,15 @@ Figure 12
-## **Summary**
+## Summary
 
 Azure key vault helps you to keep your application's secrets out of the application. You can use it to isolate secrets from your code files. These secrets include connection strings, API keys, environment variables, etc. You can take advantage of Azure Key Vault to keep secrets out of source control or out of your application in a centralized storage place. In this article we've taken advantage o ...
```
Changes: In this article`,` we've `used` managed identity to connect an Azure web app in .NET to Azure Key Vault and retrieve secret value from there.

joydipkanjilal replied:
Fixed, thanks!

raghunath-r-a reviewed (changes requested):
Hi @joydipkanjilal 

We have added comments/suggestions to improve the content presentation. Please consider them and make the necessary changes. 

Thank you,
@raghunath-r-a

joydipkanjilal commented on content/working-with-azure-key-vault-in-csharp/index.md:
```diff
@@ -1,14 +1,14 @@
-## **Introduction**
+## Introduction
 
 Azure Key Vault is a highly secure, dependable, and simple method to store your keys and secrets in the cloud. This article talks about Azure Key Vault, why it is useful, how to set up and configure it, and then shows how one can read a secret value stored inside it in C\#.
```
Done, thanks!

joydipkanjilal commented on content/working-with-azure-key-vault-in-csharp/index.md:
```diff
@@ -1,14 +1,14 @@
+## Introduction
 
 Azure Key Vault is a highly secure, dependable, and simple method to store your keys and secrets in the cloud. This article talks about Azure Key Vault, why it is useful, how to set up and configure it, and then shows how one can read a secret value stored inside it in C\#.
 
```
Done, thanks!

joydipkanjilal commented on content/working-with-azure-key-vault-in-csharp/index.md:
```diff
@@ -56,14 +56,14 @@ To execute the code examples provided in this article you should have each of th
-## **What is Azure Key Vault?**
+## What is Azure Key Vault?
 
 Azure Key Vault is a cloud service that enables you to store your application's secrets in a safe and secure place. You can keep the keys, passwords, certificates, and other secrets safely and securely. Azure Key Vault is a service that lets you store secrets and other sensitive configuration data for a particular application. You can take advantage of Azure Key Vault to set parameters shared amo ...
```
Updated the text. Thanks!

joydipkanjilal commented on content/working-with-azure-key-vault-in-csharp/index.md:
````diff
@@ -1,14 +1,14 @@
 ```
 ---
-title: **Working with Azure Key Vault in C\#**
+title: "Working with Azure Key Vault in C\#"
````
No, there is no specific meaning.

joydipkanjilal commented on content/working-with-azure-key-vault-in-csharp/index.md:
```diff
@@ -1,14 +1,14 @@
+title: "Working with Azure Key Vault in C\#"
 date: "2020-06-11"
 coverImage: "coverImage.png"
 author: "Joydip Kanjilal"
```
I've added the tags in the format mentioned here.

joydipkanjilal commented:
> 
> 
> Hi @joydipkanjilal
> 
> We have added comments/suggestions to improve the content presentation. Please consider them and make the necessary changes.
> 
> Thank you,
> @raghunath-r-a

Thanks for the comments - I've addressed all of them now.
